From 357ed913b78787aa2e2f605317994ad80547e1f2 Mon Sep 17 00:00:00 2001
From: zhangqing <zhangqing@rock-chips.com>
Date: Wed, 9 Sep 2015 08:42:38 -0700
Subject: [PATCH] Pmic: rk808: Move regulator init parameters from rk808.dtsi
 to board dts

Move rk808 regulator init parameters to the rk3xx-xx.dts
Convenient for different products setting and modify

Change-Id: I03673f4dbd478df0630e6b982e72597ade14d810
Signed-off-by: zhangqing <zhangqing@rock-chips.com>
---
 arch/arm/boot/dts/rk3188-tb.dts     | 72 +++++++++++++++++++++++++
 arch/arm/boot/dts/rk3288-box.dts    | 80 +++++++++++++++++++++++++++
 arch/arm/boot/dts/rk3288-chrome.dts | 80 +++++++++++++++++++++++++++
 arch/arm/boot/dts/rk3288-p977.dts   | 80 +++++++++++++++++++++++++++
 arch/arm/boot/dts/rk3288-tb.dts     | 84 ++++++++++++++++++++++++++++-
 arch/arm/boot/dts/rk3288-tesco.dts  | 72 +++++++++++++++++++++++++
 arch/arm/boot/dts/rk808.dtsi        | 80 ---------------------------
 7 files changed, 466 insertions(+), 82 deletions(-)

diff --git a/arch/arm/boot/dts/rk3188-tb.dts b/arch/arm/boot/dts/rk3188-tb.dts
index ae319eb725c0..d091770f6b48 100644
--- a/arch/arm/boot/dts/rk3188-tb.dts
+++ b/arch/arm/boot/dts/rk3188-tb.dts
@@ -615,14 +615,32 @@
 		
 		rk808_dcdc1_reg: regulator@0{
 			regulator-name= "vdd_arm";	
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc2_reg: regulator@1 {
 			regulator-name= "vdd_logic";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;		
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc3_reg: regulator@2 {
@@ -631,6 +649,13 @@
 			regulator-max-microvolt = <1200000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <1200000>;
+			};
 		};
 
 		rk808_dcdc4_reg: regulator@3 {
@@ -639,6 +664,13 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo1_reg: regulator@4 {
@@ -647,6 +679,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo2_reg: regulator@5 {
@@ -655,6 +692,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo3_reg: regulator@6 {
@@ -663,6 +705,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo4_reg:regulator@7 {
@@ -671,6 +718,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo5_reg: regulator@8 {
@@ -679,6 +731,11 @@
 			regulator-max-microvolt = <2800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo6_reg: regulator@9 {
@@ -687,6 +744,11 @@
 			regulator-max-microvolt = <1200000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1200000>;
+			};
 		};
 
 		rk808_ldo7_reg: regulator@10 {
@@ -695,6 +757,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo8_reg: regulator@11 {
@@ -703,6 +770,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/rk3288-box.dts b/arch/arm/boot/dts/rk3288-box.dts
index 2001d2178b48..5502913bc477 100644
--- a/arch/arm/boot/dts/rk3288-box.dts
+++ b/arch/arm/boot/dts/rk3288-box.dts
@@ -766,14 +766,32 @@
 
 		rk808_dcdc1_reg: regulator@0{
 			regulator-name= "vdd_arm";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc2_reg: regulator@1 {
 			regulator-name= "vdd_gpu";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc3_reg: regulator@2 {
@@ -782,6 +800,13 @@
 			regulator-max-microvolt = <1200000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <1200000>;
+			};
 		};
 
 		rk808_dcdc4_reg: regulator@3 {
@@ -790,6 +815,13 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
     /* NO USED, 3.3V*/
@@ -799,6 +831,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
     /* BOX:RK1000s, 3.3V  */
@@ -808,6 +845,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
     /* RK3288 PLL,USB PHY, 1.0V */
@@ -817,6 +859,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
     /* BOX:RK1000S CORE, 1.8V  */
@@ -826,6 +873,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
     /* SDMMC IO, 3.3V*/
@@ -835,6 +887,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
     /* CAMERA, 1.8V  box modify*/
@@ -844,6 +901,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
     /* RK3288 USB PHY, SAR-ADC, WIFI IO, 1.8V */
@@ -853,6 +915,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
     /* DTV, 3.3V  box modify*/
@@ -862,18 +929,31 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo9_reg: regulator@12 {
 			regulator-name= "rk_ldo9";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+			};
 		};
 
 		rk808_ldo10_reg: regulator@13 {
 			regulator-name= "rk_ldo10";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+			};
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/rk3288-chrome.dts b/arch/arm/boot/dts/rk3288-chrome.dts
index d50420099480..5b6723f2c958 100755
--- a/arch/arm/boot/dts/rk3288-chrome.dts
+++ b/arch/arm/boot/dts/rk3288-chrome.dts
@@ -647,14 +647,32 @@
 		
 		rk808_dcdc1_reg: regulator@0{
 			regulator-name= "vdd_arm";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc2_reg: regulator@1 {
 			regulator-name= "vdd_gpu";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc3_reg: regulator@2 {
@@ -663,6 +681,13 @@
 			regulator-max-microvolt = <1200000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <1200000>;
+			};
 		};
 
 		rk808_dcdc4_reg: regulator@3 {
@@ -671,6 +696,13 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo1_reg: regulator@4 {
@@ -679,6 +711,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo2_reg: regulator@5 {
@@ -687,6 +724,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo3_reg: regulator@6 {
@@ -695,6 +737,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo4_reg:regulator@7 {
@@ -703,6 +750,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo5_reg: regulator@8 {
@@ -711,6 +763,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo6_reg: regulator@9 {
@@ -719,6 +776,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo7_reg: regulator@10 {
@@ -727,6 +789,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo8_reg: regulator@11 {
@@ -735,18 +802,31 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo9_reg: regulator@12 {
 			regulator-name= "rk_ldo9";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+			};
 		};
 
 		rk808_ldo10_reg: regulator@13 {
 			regulator-name= "rk_ldo10";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+			};
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index c567b687e2f9..9818d9dcd536 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -637,14 +637,32 @@
 		
 		rk808_dcdc1_reg: regulator@0{
 			regulator-name= "vdd_arm";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc2_reg: regulator@1 {
 			regulator-name= "vdd_gpu";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc3_reg: regulator@2 {
@@ -653,6 +671,13 @@
 			regulator-max-microvolt = <1200000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <1200000>;
+			};
 		};
 
 		rk808_dcdc4_reg: regulator@3 {
@@ -661,6 +686,13 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo1_reg: regulator@4 {
@@ -669,6 +701,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo2_reg: regulator@5 {
@@ -677,6 +714,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo3_reg: regulator@6 {
@@ -685,6 +727,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo4_reg:regulator@7 {
@@ -693,6 +740,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo5_reg: regulator@8 {
@@ -701,6 +753,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo6_reg: regulator@9 {
@@ -709,6 +766,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo7_reg: regulator@10 {
@@ -717,6 +779,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo8_reg: regulator@11 {
@@ -725,18 +792,31 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo9_reg: regulator@12 {
 			regulator-name= "rk_ldo9";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+			};
 		};
 
 		rk808_ldo10_reg: regulator@13 {
 			regulator-name= "rk_ldo10";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+			};
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/rk3288-tb.dts b/arch/arm/boot/dts/rk3288-tb.dts
index 855f0fdfc522..c8cc6c30a6a4 100755
--- a/arch/arm/boot/dts/rk3288-tb.dts
+++ b/arch/arm/boot/dts/rk3288-tb.dts
@@ -900,14 +900,32 @@
 		
 		rk808_dcdc1_reg: regulator@0{
 			regulator-name= "vdd_arm";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc2_reg: regulator@1 {
 			regulator-name= "vdd_gpu";
-			//regulator-always-on;
-			//regulator-boot-on;
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
+			/*regulator-always-on;*/
+			/*regulator-boot-on;*/
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc3_reg: regulator@2 {
@@ -916,6 +934,13 @@
 			regulator-max-microvolt = <1200000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <1200000>;
+			};
 		};
 
 		rk808_dcdc4_reg: regulator@3 {
@@ -924,6 +949,13 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo1_reg: regulator@4 {
@@ -932,6 +964,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo2_reg: regulator@5 {
@@ -940,6 +977,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo3_reg: regulator@6 {
@@ -948,6 +990,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo4_reg:regulator@7 {
@@ -956,6 +1003,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo5_reg: regulator@8 {
@@ -964,6 +1016,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo6_reg: regulator@9 {
@@ -972,6 +1029,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo7_reg: regulator@10 {
@@ -980,6 +1042,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo8_reg: regulator@11 {
@@ -988,18 +1055,31 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo9_reg: regulator@12 {
 			regulator-name= "rk_ldo9";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+			};
 		};
 
 		rk808_ldo10_reg: regulator@13 {
 			regulator-name= "rk_ldo10";
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+			};
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/rk3288-tesco.dts b/arch/arm/boot/dts/rk3288-tesco.dts
index 50c0a14fc5b4..235962b0b0b7 100755
--- a/arch/arm/boot/dts/rk3288-tesco.dts
+++ b/arch/arm/boot/dts/rk3288-tesco.dts
@@ -606,14 +606,32 @@
 		
 		rk808_dcdc1_reg: regulator@0{
 			regulator-name= "vdd_arm";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc2_reg: regulator@1 {
 			regulator-name= "vdd_gpu";
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1500000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-disabled;
+				regulator-state-uv = <900000>;
+			};
 		};
 
 		rk808_dcdc3_reg: regulator@2 {
@@ -622,6 +640,13 @@
 			regulator-max-microvolt = <1200000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <1200000>;
+			};
 		};
 
 		rk808_dcdc4_reg: regulator@3 {
@@ -630,6 +655,13 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-mode = <0x2>;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-mode = <0x2>;
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo1_reg: regulator@4 {
@@ -638,6 +670,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo2_reg: regulator@5 {
@@ -646,6 +683,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 
 		rk808_ldo3_reg: regulator@6 {
@@ -654,6 +696,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo4_reg:regulator@7 {
@@ -662,6 +709,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo5_reg: regulator@8 {
@@ -670,6 +722,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <2800000>;
+			};
 		};
 
 		rk808_ldo6_reg: regulator@9 {
@@ -678,6 +735,11 @@
 			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-disabled;
+				regulator-state-uv = <1000000>;
+			};
 		};
 
 		rk808_ldo7_reg: regulator@10 {
@@ -686,6 +748,11 @@
 			regulator-max-microvolt = <1800000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <1800000>;
+			};
 		};
 
 		rk808_ldo8_reg: regulator@11 {
@@ -694,6 +761,11 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 			regulator-boot-on;
+			regulator-initial-state = <3>;
+			regulator-state-mem {
+				regulator-state-enabled;
+				regulator-state-uv = <3300000>;
+			};
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/rk808.dtsi b/arch/arm/boot/dts/rk808.dtsi
index f536222a3178..54cd24d703d9 100755
--- a/arch/arm/boot/dts/rk808.dtsi
+++ b/arch/arm/boot/dts/rk808.dtsi
@@ -9,151 +9,71 @@
 		rk808_dcdc1_reg: regulator@0 {
 			reg = <0>;
 			regulator-compatible = "rk_dcdc1";
-			regulator-min-microvolt = <700000>;
-			regulator-max-microvolt = <1500000>;
-			regulator-initial-mode = <0x2>;
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-mode = <0x2>;
-				regulator-state-disabled;//disabled
-				regulator-state-uv = <900000>;
-			};		
 		};
 
 		rk808_dcdc2_reg: regulator@1 {
 			reg = <1>;
 			regulator-compatible = "rk_dcdc2";
-			regulator-min-microvolt = <700000>;
-			regulator-max-microvolt = <1500000>;	
-			regulator-initial-mode = <0x2>;
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-mode = <0x2>;
-				regulator-state-disabled;
-				regulator-state-uv = <900000>;
-			};		
 		};
 
 		rk808_dcdc3_reg: regulator@2 {
 			reg = <2>;
 			regulator-compatible = "rk_dcdc3";
-			regulator-initial-mode = <0x2>;
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-mode = <0x2>;
-				regulator-state-enabled;
-				regulator-state-uv = <1200000>;
-			};	
 		};
 
 		rk808_dcdc4_reg: regulator@3 {
 			reg = <3>;
 			regulator-compatible = "rk_dcdc4";
-			regulator-initial-mode = <0x2>;
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-mode = <0x2>;
-				regulator-state-enabled;
-				regulator-state-uv = <2800000>;
-			};		
 		};
 
 		rk808_ldo1_reg: regulator@4 {
 			reg = <4>;
 			regulator-compatible = "rk_ldo1";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-enabled;
-				regulator-state-uv = <3300000>;
-			};	
 		};
 
 		rk808_ldo2_reg: regulator@5 {
 			reg = <5>;
 			regulator-compatible = "rk_ldo2";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-enabled;
-				regulator-state-uv = <3300000>;
-			};	
 		};
 
 		rk808_ldo3_reg: regulator@6 {
 			reg = <6>;
 			regulator-compatible = "rk_ldo3";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-enabled;
-				regulator-state-uv = <1000000>;
-			};	
 		};
 
 		rk808_ldo4_reg: regulator@7{
 			reg = <7>;
 			regulator-compatible = "rk_ldo4";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-disabled;
-				regulator-state-uv = <1800000>;
-			};	
 		};
 
 		rk808_ldo5_reg: regulator@8{
 			reg = <8>;
 			regulator-compatible = "rk_ldo5";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-enabled;
-				regulator-state-uv = <2800000>;
-			};	
 		};
 
 		rk808_ldo6_reg: regulator@9{
 			reg = <9>;
 			regulator-compatible = "rk_ldo6";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-disabled;
-				regulator-state-uv = <1000000>;
-			};	
 		};
 
 		rk808_ldo7_reg: regulator@10 {
 			reg = <10>;
 			regulator-compatible = "rk_ldo7";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-enabled;
-				regulator-state-uv = <1800000>;
-			};	
 		};
 
 		rk808_ldo8_reg: regulator@11{
 			reg = <11>;
 			regulator-compatible = "rk_ldo8";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-enabled;
-				regulator-state-uv = <3300000>;
-			};	
 		};
 		
 		rk808_ldo9_reg: regulator@12{
 			reg = <12>;
 			regulator-compatible = "rk_ldo9";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-enabled;
-			};	
 		};
 		
 		rk808_ldo10_reg: regulator@13{
 			reg = <13>;
 			regulator-compatible = "rk_ldo10";
-			regulator-initial-state = <3>;
-			regulator-state-mem {
-				regulator-state-disabled;
-			};	
 		};
 
 	};
-- 
2.35.3

